common/org.wso2.developerstudio.eclipse.utils/src/org/wso2/developerstudio/eclipse/utils/wst/Axis2ServiceUtils.java

```diff
@@ -39,166 +35,301 @@
 import org.eclipse.core.resources.ResourcesPlugin;
 import org.eclipse.core.runtime.CoreException;
 import org.w3c.dom.Document;
+import org.w3c.dom.NamedNodeMap;
 import org.w3c.dom.Node;
 import org.w3c.dom.NodeList;
+import org.wso2.developerstudio.eclipse.logging.core.IDeveloperStudioLog;
+import org.wso2.developerstudio.eclipse.logging.core.Logger;
+import org.wso2.developerstudio.eclipse.utils.Activator;
+import org.wso2.developerstudio.eclipse.utils.exception.Axis2ServiceUtilsException;
 import org.wso2.developerstudio.eclipse.utils.file.FileUtils;
 import org.xml.sax.SAXException;
 
+/**
+ * This class provides static utility methods for Axis2 service project management tasks.
+ * Such as Axis2 services.xml file manipulation, data extraction etc.
+ */
 public class Axis2ServiceUtils {
-	public static Map<IFolder,IProject> getServiceFolders() throws CoreException{
+	private static final IDeveloperStudioLog log = Logger.getLog(Activator.PLUGIN_ID);
+
+	private static final String SERVICES_XML_FILE_NAME = "services.xml";
+	private static final String META_INF_FOLDER_NAME = "META-INF";
+	private static final String SERVICE_ELEMENT_NAME = "service";
+	private static final String PARAMETER_ELEMENT_NAME = "parameter";
+	private static final String NAME_ATTRIBUTE = "name";
+	private static final String SERVICE_CLASS_ATTRIBUTE_VALUE = "ServiceClass";
+	private static final String DOT_SEPARATOR = ".";
+
+	/**
+	 * This method provides logic for getting all Axis2 service folders with related project in workspace.
+	 *
+	 * @return a map contains all service folders with corresponding project
+	 * @throws org.wso2.developerstudio.eclipse.utils.exception.Axis2ServiceUtilsException
+	 */
+	public static Map<IFolder, IProject> getServiceFolders() throws Axis2ServiceUtilsException {
 		HashMap<IFolder, IProject> map = new HashMap<IFolder, IProject>();
-		for(IProject project:ResourcesPlugin.getWorkspace().getRoot().getProjects()){
-			if (WebUtils.isDynamicWebProject(project)){
+		for (IProject project : ResourcesPlugin.getWorkspace().getRoot().getProjects()) {
+			if (WebUtils.isDynamicWebProject(project)) {
 				final IFolder servicesFolder;
-                try {
-	                servicesFolder = WebUtils.getAxis2WebContainerWEB_INFServicesFolderPath(project);
-                } catch (Exception e1) {
```

Comment:
Seems like there's no exception thrown in this line. Better to correct and test this.